File: balance.js
const readlineSync = require('readline-sync');
const fs = require('fs');
const utils = require('./utils');

function checkBalance() {
    const accountName = readlineSync.question('Entrez le nom du compte: ');

    const transactions = fs.readFileSync(utils.getTransactionFilePath(), 'utf8');
    const walletInfo = utils.calculateBalance(transactions, accountName);

    if (walletInfo !== undefined) {
        console.log(`Solde actuel du compte "${accountName}":`);
        console.log(`- Argent: ${walletInfo.argent}`);
        console.log(`- Cartes Bancaires: ${walletInfo.cartesBancaires}`);
        console.log(`- Permis de Conduire: ${walletInfo.permisDeConduire}`);
        console.log(`- Photos d'Identité: ${walletInfo.photosIdentite}`);
        console.log(`- Cartes de Visite: ${walletInfo.cartesVisite}`);
    } else {
        console.log(`Le compte "${accountName}" n'existe pas.`);
    }
}

module.exports = {
    checkBalance
};

File: Balance.cs
using System;
using System.IO;

class Balance
{
    private readonly Utils _utils;

    public Balance(Utils utils)
    {
        _utils = utils;
    }

    public void CheckBalance()
    {
        Console.Write("Entrez le nom du compte : ");
        string accountName = Console.ReadLine();

        string transactions = File.ReadAllText(_utils.GetTransactionFilePath());
        WalletInfo walletInfo = _utils.CalculateBalance(transactions, accountName);

        if (walletInfo != null)
        {
            Console.WriteLine($"Solde actuel du compte \"{accountName}\":");
            Console.WriteLine($"- Argent: {walletInfo.Argent}");
            Console.WriteLine($"- Cartes Bancaires: {walletInfo.CartesBancaires}");
            Console.WriteLine($"- Permis de Conduire: {walletInfo.PermisDeConduire}");
            Console.WriteLine($"- Photos d'Identité: {walletInfo.PhotosIdentite}");
            Console.WriteLine($"- Cartes de Visite: {walletInfo.CartesVisite}");
        }
        else
        {
            Console.WriteLine($"Le compte \"{accountName}\" n'existe pas.");
        }
    }
}
